miniapps/shifted/diffusion.cpp

```diff
@@ -0,0 +1,431 @@
+﻿// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+//
+//      -----------------------------------------------------------------
+//      Shifted Boundary Miniapp: Finite element immersed boundary solver
+//      -----------------------------------------------------------------
+//
+// TODO - problem and approach descriptions, references.
+//
+// Compile with: make diffusion
+//
+// Sample runs:
+//   Problem 1: Circular hole of radius 0.2 at the center of the domain.
+//              Solves -nabla^u = 1 with homogeneous boundary conditions.
+//   mpirun -np 4 diffusion -m ../../data/inline-quad.mesh -rs 3 -o 1 -vis -lst 1
+//   mpirun -np 4 diffusion -m ../../data/inline-hex.mesh -rs 2 -o 2 -vis -lst 1 -ho 1 -alpha 10
+//
+//   Problem 2: Circular hole of radius 0.2 at the center of the domain.
+//              Solves -nabla^u = f with inhomogeneous boundary conditions, and
+//              f is setup such that u = x^p + y^p, where p = 2 by default.
+//              This is a 2D convergence test.
+//   mpirun -np 4 diffusion -rs 2 -o 2 -vis -lst 2
```

Comment:
Could you add more details how to run the convergence tests. I don't think simply changing `-rs` works.